x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/schema_meta_engine_logic.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { SchemaBaseLogic, SchemaBaseValues, SchemaBaseActions } from './schema_base_logic';
+import { MetaEngineSchemaApiResponse } from './types';
+
+interface MetaEngineSchemaValues extends SchemaBaseValues {
+  fields: MetaEngineSchemaApiResponse['fields'];
+  conflictingFields: MetaEngineSchemaApiResponse['conflictingFields'];
+  conflictingFieldsCount: number;
+  hasConflicts: boolean;
+}
+
+interface MetaEngineSchemaActions extends SchemaBaseActions {
+  loadMetaEngineSchema(): void;
+  onMetaEngineSchemaLoad(response: MetaEngineSchemaApiResponse): MetaEngineSchemaApiResponse;
+}
+
+export const MetaEngineSchemaLogic = kea<
+  MakeLogicType<MetaEngineSchemaValues, MetaEngineSchemaActions>
+>({
+  path: ['enterprise_search', 'app_search', 'meta_engine_schema_logic'],
+  connect: {
+    values: [SchemaBaseLogic, ['dataLoading', 'schema']],
+    actions: [SchemaBaseLogic, ['loadSchema', 'setSchema']],
+  },
+  actions: {
+    loadMetaEngineSchema: true,
+    onMetaEngineSchemaLoad: (response) => response,
```

Comment:
Building on my previous comments, I think this could be replaced with an `onSchemaLoad` from the base logic.

Reply:
https://github.com/elastic/kibana/pull/99548/commits/82a711c79f3c9a4cc8de48a06578f7480a2aa4f3